[github-actions] test_netdata_publisher.py fails intermittently

[logs_91695.zip](https://github.com/openthread/openthread/files/6980058/logs_91695.zip)
[packet-verification-1.1-on-1.2-pcaps.zip](https://github.com/openthread/openthread/files/6980059/packet-verification-1.1-on-1.2-pcaps.zip)

```
======================================================================
FAIL: test (__main__.NetDataPublisher)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./tests/scripts/thread-cert/test_netdata_publisher.py", line 425, in test
    self.assertTrue(rlocs.count(end_dev1.get_addr16()) == 1)
AssertionError: False is not true

----------------------------------------------------------------------
```

--- a/tests/scripts/thread-cert/test_netdata_publisher.py
+++ b/tests/scripts/thread-cert/test_netdata_publisher.py
@@ -416,7 +416,7 @@ class NetDataPublisher(thread_cert.TestCase):
 
         for node in routers:
             node.netdata_publish_prefix(ON_MESH_PREFIX, ON_MESH_FLAGS, 'med')
-        self.simulator.go(WAIT_TIME)
+            self.simulator.go(WAIT_TIME)
         prefixes = leader.get_prefixes()
         self.check_num_of_prefixes(prefixes, 0, 1 + len(routers), 0)
         self.assertTrue(1 + len(routers) >= DESIRED_NUM_ON_MESH_PREFIX)

